### DISCUSSION TITLE
Mocking a Wallet Connection in a Wagmi + Next.js App

### DISCUSSION THREAD

**Discussion by frankfka** [wevm/wagmi#4438] [Q&A]
Hey all,

We have a usecase where we have the private keys to a wallet, and we want to "mock" the connection (i.e connect the wallet as if the user has connected via a normal connector).

We would essentially want to set `wagmi` to the "connected" state with all the relevant information associated with this wallet.

Is there some sort of "WalletConnector" that is available? Or is there another suggested way of doing this?

Thanks!

**Comment by frankfka** [comment]
@tmm I would think that this would be some variation of `MockConnector`, do you have any suggestions on how best to approach this?

Thank you!

**Comment by frankfka** [comment]
Where I'm at now is:

```ts
async getProvider({ chainId } = {}) {
  const chain =
    config.chains.find((x) => x.id === chainId) ?? config.chains[0];

  const walletClient = createWalletClient({
    account,
    chain,
    transport: http(),
  });
  return custom({ request: walletClient.request })({ retryCount: 0 });
},
```

Everything else seems to work, but using `privateKeyToAccount` and an `http` transport doesn't seem to work with EIP1193 requests. Does viem/wagmi have an adapter that provides an EIP1193 compatible layer on top of a private key account, or do we have to implement this ourselves?

**Reply by tmm** [reply]
You can implement it yourself. Some examples:

- [Wagmi `mock` connector](https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/mock.ts#L137-L276)
- [Porto `provider`](https://github.com/ithacaxyz/porto/blob/main/src/internal/provider.ts#L44-L518)
